### PR TITLE
look for is.unsorted() behavior in sort_linter

### LINKED ISSUE #2072: is_unsorted: new linter, or part of sort_linter() ?
One linter for #884 looks for `sort(x) == x` and recommends `!is.unsorted(x)`.

Should this be a new linter (e.g. `is_unsorted_linter()`), or subsumed into the recent `sort_linter()`?

### PR DESCRIPTION
**Pull request by MichaelChirico**
Closes #2072, part of #884

### CODE REVIEW

File: tests/testthat/test-sort_linter.R [hunk 1]
```diff
@@ -79,3 +79,36 @@ test_that("sort_linter works with multiple lints in a single expression", {
   )
 
 })
+
+test_that("sorted_linter skips usages calling sort arguments", {
```
Comment (AshesITR) [inline tests/testthat/test-sort_linter.R:83]: sort_linter ...

**Review by AshesITR — DISMISSED** [review]

**Comment by MichaelChirico** [comment]
Ah that's annoying.

 - Merging in the Web UI doesn't dismiss a review _unless_ there's a conflict
 - Merging locally & pushing _does_ dismiss a review, even if it was a clean merge
 - The Web UI ignores our .gitattributes

So I can cleanly merge a simple NEWS update locally, but not on the web without dismissing the review. Alas.

File: R/sort_linter.R [hunk 2]
```diff
@@ -1,9 +1,24 @@
...
 #' [sort()] is the dedicated option to sort a list or vector. It is more legible
 #' and around twice as fast as `.[order(.)]`, with the gap in performance
 #' growing with the vector size.
 #'
+#' Second, it requires usage of [is.unsorted()] over equivalents using `sort()`.
+#'
+#' The base function `is.unsorted()` exists to test the sortedness of a vector.
+#'   Prefer it to inefficient and less-readable equivalents like
+#'   `x != sort(x)`. The same goes for checking `x == sort(x)` -- use
+#'   `!is.unsorted(x)` instead.
```
Comment (IndrajeetPatil) [inline R/sort_linter.R:16]: Do you think the lint message should reflect the exact replacement, depending on whether it's a regular (`is.unsorted()`) or inverted (`!is.unsorted(x)`) comparison? Or will that be an overkill?

File: NEWS.md [hunk 3]
```diff
@@ -23,6 +23,7 @@
   + `yoda_test_linter()`
 * `sprintf_linter()` is pipe-aware, so that `x %>% sprintf(fmt = "%s")` no longer lints (#1943, @MichaelChirico).
 * `line_length_linter()` helpfully includes the line length in the lint message (#2057, @MichaelChirico).
+* `sort_linter()` checks for code like `x == sort(x)` which is better served by using the function `is.unsorted()` (part of #884, @MichaelChirico).
```
Comment (IndrajeetPatil) [inline NEWS.md:26]: This is pretty cool! I had no idea that this function existed 🙈

File: R/sort_linter.R (hunk 2 above)
Comment (MichaelChirico) [reply R/sort_linter.R:16]: The same thought crossed my mind. I think it's a good call b/c sometimes I have to pause a minute and remember which is the equivalent. It's better to write it in the message (and trivial to do so). Thanks!

**Review by AshesITR — APPROVED** [review]

**Pull request merged**